miner/taiko_worker.go

```diff
@@ -72,47 +83,46 @@ func (w *Miner) buildTransactionsLists(
 		localTxs, remoteTxs = w.getPendingTxs(localAccounts, baseFee)
 	)
 
-	commitTxs := func(firstTransaction *types.Transaction) (*types.Transaction, *PreBuiltTxList, error) {
+	commitTxs := func(pruningResult *txsPruningResult) (*txsPruningResult, *PreBuiltTxList, error) {
 		env.tcount = 0
 		env.txs = []*types.Transaction{}
-		env.gasPool = new(core.GasPool).AddGas(blockMaxGasLimit)
+		env.gasPool = new(core.GasPool).AddGas(blockMaxGasLimit - accumulateGasUsed(pruningResult.ReceiptsPruned))
 		env.header.GasLimit = blockMaxGasLimit
 
-		lastTransaction := w.commitL2Transactions(
+		txsPruningResult, err := w.commitL2Transactions(
```

Comment:
Shadow declaration of `txsPruningResult `.

Reply:
At least there is no lint warning, so actually it's ok?

Reply:
But anyway renamed